app/server/api/disbursement_api.py

```diff
@@ -139,20 +140,16 @@ def post(self):
             disbursement_amount = disbursement_amount,
             transfer_type = transfer_type
         )
-
         if include_accounts:
-            transfer_accounts = db.session.query(TransferAccount).filter(TransferAccount.id.in_(include_accounts)).all()
+            transfer_accounts = db.session.query(TransferAccount).filter(TransferAccount.id.in_(include_accounts)).options(load_only("id")).all()
         else:
-            search_query = generate_search_query(search_string, filters, order, sort_by_arg, include_user=True)
+            search_query = generate_search_query(search_string, filters, order, sort_by_arg, include_user=False)
             search_query = search_query.filter(TransferAccount.id.notin_(exclude_accounts))
-            results = search_query.all()
-            transfer_accounts = [r[0] for r in results] # Get TransferAccount (TransferAccount, searchRank, User)
-        d.transfer_accounts.extend(transfer_accounts)
-
-        db.session.flush()
-
+            transfer_accounts = search_query.options(load_only("id")).all()
```

Comment:
No need to load the full-fat transfer accounts, all we need is IDs to use as foreign keys